### ISSUE TITLE
FreezeMaskedDivRemPass: freeze is insufficient, LLVM exploits sdiv-by-zero UB

### ISSUE THREAD

**Issue by etiotto** [intel/intel-xpu-backend-for-triton#6674]
## Problem

When `TRITON_INTEL_PREDICATED_LOAD=0`, Triton masked loads lower to conditional blocks that produce phi nodes with a zero default on the false path:

```llvm
  br i1 %mask, label %load_bb, label %merge
load_bb:
  %val = load i64, ptr addrspace(1) %ptr
  br label %merge
merge:
  %phi = phi i64 [ %val, %load_bb ], [ 0, %entry ]
  %res = sdiv i64 %x, %phi          ; UB when %mask is false
```

When the phi is used as a divisor in `sdiv`/`srem`, the zero default makes `sdiv x, 0` undefined behavior. LLVM's SimplifyCFG exploits this UB to insert `llvm.assume(%mask)`, which propagates and corrupts unrelated operations — for example, making predicated stores unconditional and second loads unguarded. This produces incorrect results from legal Triton kernels.

### Why `freeze` does not work

The existing `FreezeMaskedDivRemPass` inserts `freeze` between the phi and the div:

```llvm
%frozen = freeze i64 %phi
%res = sdiv i64 %x, %frozen
```

This is insufficient because **both phi inputs are well-defined** (a loaded value and the constant `0`). LLVM can prove this and legally removes the `freeze` as a no-op *before* SimplifyCFG runs, leaving the UB intact.

### Two issues

1. **What**: `freeze` is the wrong tool — it cannot prevent UB when inputs are provably well-defined.
2. **When**: The pass was registered via `registerPeepholeEPCallback`, which fires *after* InstCombine inside the O3 pipeline. By that point, SimplifyCFG has already exploited the UB and potentially folded the conditional blocks, destroying the phi nodes the pass needs to match.

## Fix

1. Replace `freeze` with `select(divisor == 0, 1, divisor)` — this eliminates the UB entirely. The zero case only arises on the false path where both dividend and divisor are zero, so `0/0` (UB) becomes `0/1 = 0`, a well-defined value that is never observed.

2. Move the pass to run *before* `buildPerModuleDefaultPipeline` so it acts before any O3 optimization (especially SimplifyCFG).

3. Rename the pass from `FreezeMaskedDivRemPass` to `GuardMaskedDivRemPass` to reflect the new approach.

### Reproducer

```python
import torch, triton, triton.language as tl, os
os.environ["TRITON_INTEL_PREDICATED_LOAD"] = "0"

@triton.jit
def simple_div_kernel(a, b, out_ptr2, out_ptr3, xnumel, XBLOCK: tl.constexpr):
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.arange(0, XBLOCK)[:]
    xmask = xindex < xnumel
    x0 = xindex
    tmp0 = tl.load(a + (x0), xmask)
    tmp2 = tl.load(b + (x0), xmask)
    tmp6 = tmp0 // tmp2
    tl.store(out_ptr2 + (x0), tmp6, xmask)

torch.manual_seed(0)
a = torch.randint(2**32, 2**40, [100, 100], device="xpu")
b = torch.randint(-10, -1, [100, 100], device="xpu")
output2 = torch.zeros_like(a)
output3 = torch.full_like(a, 42)
n_elements = output2.numel()
grid = lambda meta: (triton.cdiv(n_elements, meta['XBLOCK']), )
simple_div_kernel[grid](a, b, output2, output3, n_elements, XBLOCK=128)

corrupted = (output3 != 42).sum().item()
print(f"output3 corrupted elements: {corrupted}")  # Non-zero without fix
ref = torch.div(a, b, rounding_mode="trunc")
print(f"output2 mismatches: {(output2 - ref).abs().gt(0).sum().item()}")  # Non-zero without fix
```

## Files changed

- `third_party/intel/lib/Target/LLVMIR/LLVMIRGuardMaskedDivRem.cpp` (renamed from `LLVMIRFreezeMaskedDivRem.cpp`) — select-based guard, also handles `UDiv`/`URem`
- `third_party/intel/lib/Target/LLVMIR/LLVMPasses.h` — renamed pass class
- `third_party/intel/lib/Target/LLVMIR/CMakeLists.txt` — updated filename
- `third_party/intel/triton_xpu.cc` — moved pass before O3 pipeline
- `bin/triton-llvm-opt.cpp` — renamed CLI flag to `-guard-masked-div-rem`
- `test/LLVMIR/guard-masked-div-rem.ll` (renamed from `freeze-masked-div-rem.ll`) — updated CHECK lines for select pattern
